src/Resolvers/Microsoft.NET.Sdk.WorkloadResolver/MSBuildWorkloadSdkResolver.cs

```diff
@@ -0,0 +1,168 @@
+﻿using Microsoft.Build.Framework;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using Microsoft.NET.Sdk.WorkloadManifestReader;
+
+#if NET
+using Microsoft.DotNet.Cli;
+#else
+using Microsoft.DotNet.DotNetSdkResolver;
+#endif
+
+#nullable disable
+
+namespace Microsoft.NET.Sdk.WorkloadResolver
+{
+    public class MSBuildWorkloadSdkResolver : SdkResolver
+    {
+        public override string Name => "Microsoft.DotNet.MSBuildWorkloadSdkResolver";
+
+        public override int Priority => 4000;
+
+        private bool _enabled;
+
+        private IWorkloadManifestProvider _workloadManifestProvider;
+        private IWorkloadResolver _workloadResolver;
+
+
+#if NETFRAMEWORK
+        private readonly NETCoreSdkResolver _sdkResolver;
+#endif
+
+
+        public MSBuildWorkloadSdkResolver()
+        {
+            //  Put workload resolution behind a feature flag.
+            _enabled = false;
+            var envVar = Environment.GetEnvironmentVariable("MSBuildEnableWorkloadResolver");
+            if (envVar != null)
+            {
+                if (envVar.Equals("true", StringComparison.OrdinalIgnoreCase))
+                {
+                    _enabled = true;
+                }
+            }
+
+            if (!_enabled)
+            {
+                string sentinelPath = Path.Combine(Path.GetDirectoryName(typeof(MSBuildWorkloadSdkResolver).Assembly.Location), "EnableWorkloadResolver.sentinel");
+                if (File.Exists(sentinelPath))
+                {
+                    _enabled = true;
+                }
+            }
+
+#if NETFRAMEWORK
+            if (_enabled)
+            {
+                _sdkResolver = new NETCoreSdkResolver();
+            }
+#endif
+        }
+
+        private void InitializeWorkloadResolver(SdkResolverContext context)
+        {
+            var dotnetRootPath = GetDotNetRoot(context);
+
+            var sdkDirectory = GetSdkDirectory(context);
+            //  The SDK version is the name of the SDK directory (ie dotnet\sdk\5.0.100)
+            var sdkVersion = Path.GetFileName(sdkDirectory);
+
+            _workloadManifestProvider ??= new SdkDirectoryWorkloadManifestProvider(dotnetRootPath, sdkVersion);
+            //  TODO: Fix namespace / type names so there's not a collision for WorkloadResolver
+            _workloadResolver ??= new Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadResolver(_workloadManifestProvider, dotnetRootPath);
+        }
+
+        public override SdkResult Resolve(SdkReference sdkReference, SdkResolverContext resolverContext, SdkResultFactory factory)
+        {
+            if (!_enabled)
+            {
+                return null;
+            }
+
+            InitializeWorkloadResolver(resolverContext);
+
+            if (sdkReference.Name.Equals("Microsoft.NET.SDK.WorkloadAutoImportPropsLocator", StringComparison.OrdinalIgnoreCase))
```

Comment:
Does the test cover workload props and targets can be included to the project?

Reply:
Yes, `It_should_build_with_workload` tests that the targets are imported by using a fake target platform that the workload targets enable.  And `It_should_import_AutoImports_for_installed_workloads` checks that the props are imported via a property that is set in the .props file.